app/src/main/java/com/example/plantsapp/data/contentprovider/PlantsContentProvider.kt

```diff
@@ -20,14 +20,22 @@ class PlantsContentProvider : ContentProvider() {
     @EntryPoint
     @InstallIn(SingletonComponent::class)
     interface PlantsContentProviderEntryPoint {
-        @FirebaseQualifier
-        fun plantsRepository(): PlantsRepository
+        fun getCreatePlantsCursorUseCase(): CreatePlantsCursorUseCase
+        fun getCreateTasksCursorUseCase(): CreateTasksCursorUseCase
+        fun getCreateTaskHistoryCursorCase(): CreateTaskHistoryCursorUseCase
     }
 
     private lateinit var hiltEntryPoint: PlantsContentProviderEntryPoint
-    private val plantsRepository: PlantsRepository by lazy {
-        hiltEntryPoint.plantsRepository()
+    private val createPlantsCursorUseCase: CreatePlantsCursorUseCase by lazy {
+        hiltEntryPoint.getCreatePlantsCursorUseCase()
     }
+    private val createTasksCursorUseCase: CreateTasksCursorUseCase by lazy {
+        hiltEntryPoint.getCreateTasksCursorUseCase()
+    }
+    private val createTaskHistoryCursorUseCase: CreateTaskHistoryCursorUseCase by lazy {
+        hiltEntryPoint.getCreateTaskHistoryCursorCase()
+    }
+    private val uriMatcher = UriMatcher(UriMatcher.NO_MATCH)
```

Comment:
Probably you can create and initialize urimvatcher here, and not in  onCreate()

Reply:
What's more - just for your info, you can hide all logic to matcher to some class (do not do it, but for some cases it could be useful)